tools/build_langserver/langserver/handler.go

```diff
@@ -70,6 +71,18 @@ func (h *LsHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jsonrp
 		req.Method != "initialzed" && req.Method != "shutdown" {
 		ctx = h.requestStore.Store(ctx, req)
 		defer h.requestStore.Cancel(req.ID)
+
+		go func() {
+			t, err := h.diagPublisher.queue.Get(1)
```

Comment:
I would probably stick most of this into a method on `diagPublisher` rather than directly accessing its queue here. Keeps the code here shorter but also makes it easier for it to manage things.